ci/Dockerfile

```diff
@@ -223,7 +223,8 @@ RUN export DEAL_II_HASH=1a364b58c4a5e2235c1282544d55486d41bead96 && \
         -DDEAL_II_WITH_ARPACK=ON \
         -DARPACK_DIR=${ARPACK_DIR} \
         -DDEAL_II_WITH_CUDA=ON \
-        -DDEAL_II_WITH_CXX14=OFF \
+        -DDEAL_II_WITH_CXX14=ON \
+        -DDEAL_II_WITH_CXX17=OFF \
```

Comment:
Do we actually need the `DEAL_II_WITH_CXX17=OFF`?

Reply:
It should be off by default but this way it is obvious that it is. No version of cuda support c++17.